pkg/genericapiserver/genericapiserver.go

```diff
@@ -58,6 +58,7 @@ import (
 
 const (
 	DefaultEtcdPathPrefix = "/registry"
+	globalTimeout         = 30 * time.Second
```

Comment:
Global timeout must be longer than any downsteam timeout. So this should stay a minute (or get longer) and the componentstatus timeout should be reduced, I would say to 10-20s.